testsupport/wait/member.go

```diff
@@ -880,6 +880,69 @@ func (a *MemberAwaitility) WaitForMemberStatus(criteria ...MemberStatusWaitCrite
 	return err
 }
 
+// GetMemberOperatorConfig returns MemberOperatorConfig instance, nil if not found
+func (a *MemberAwaitility) GetMemberOperatorConfig() *toolchainv1alpha1.MemberOperatorConfig {
+	config := &toolchainv1alpha1.MemberOperatorConfig{}
+	if err := a.Client.Get(context.TODO(), test.NamespacedName(a.Namespace, "config"), config); err != nil {
+		if errors.IsNotFound(err) {
+			return nil
+		}
+		require.NoError(a.T, err)
+	}
+	return config
+}
+
+// MemberOperatorConfigWaitCriterion a function to check that an MemberOperatorConfig has the expected criteria
+type MemberOperatorConfigWaitCriterion func(*MemberAwaitility, *toolchainv1alpha1.MemberOperatorConfig) bool
+
+// UntilMemberConfigMatches returns a `MemberOperatorConfigWaitCriterion` which checks that the given
+// MemberOperatorConfig matches the provided one
+func UntilMemberConfigMatches(expectedMemberOperatorConfig *toolchainv1alpha1.MemberOperatorConfig) MemberOperatorConfigWaitCriterion {
+	return func(a *MemberAwaitility, memberConfig *toolchainv1alpha1.MemberOperatorConfig) bool {
+		return hasMatchingMemberOperatorConfig(a.T, expectedMemberOperatorConfig, memberConfig)
+	}
+}
+
+func hasMatchingMemberOperatorConfig(t *testing.T, expectedMemberOperatorConfig *toolchainv1alpha1.MemberOperatorConfig, memberConfig *toolchainv1alpha1.MemberOperatorConfig) bool {
+	if reflect.DeepEqual(expectedMemberOperatorConfig, memberConfig) {
+		t.Logf("the MemberOperatorConfig matches the expected configuration")
+	}
+	t.Logf("waiting for MemberOperatorConfig to be synced. Actual: '%+v'; Expected: '%+v'", memberConfig, expectedMemberOperatorConfig)
+	return false
+}
```

Comment:
the same question - couldn't it be part of one function `UntilMemberConfigMatches`?

Reply:
Made this a single func too: ef269aeb7f7686ab2cc1992474c672dea13271a9